target_actionnetwork/sinks.py

```diff
@@ -1,15 +1,74 @@
 """ActionNetwork target sink class, which handles writing streams."""
-from __future__ import annotations
 
 from target_actionnetwork.client import ActionNetworkSink
 
 
+class CreateAdvocacyCampaignException(Exception):
+    pass
+
+
+class CreateOutreachException(Exception):
+    pass
+
+
 class ContactsSink(ActionNetworkSink):
     """ActionNetwork target sink class for Contacts"""
 
     endpoint = "people"
     name = "Contacts"
+    advocacy_campaigns = {}
+
+    def __init__(self, target, stream_name, schema, key_properties):
+        super().__init__(target, stream_name, schema, key_properties)
+        self.get_advocacy_campaigns()
+
+    def get_advocacy_campaigns(self):
+        response = self._request("GET", "advocacy_campaigns")
+        res_json = response.json()
+        total_pages = res_json.get("total_pages")
+        current_page = 0
+        while current_page < total_pages:
+            current_page += 1
+            response = self._request("GET", "advocacy_campaigns", {"page": current_page})
+            res_json = response.json()
+            advocacy_campaigns = res_json["_embedded"]["osdi:advocacy_campaigns"]
+            for adv_camp in advocacy_campaigns:
+                if adv_camp["title"] in self.advocacy_campaigns:
+                    self.logger.warning(f"The campaign (title={adv_camp['title']}, id={adv_camp['identifiers']}) already exists, and it is mapped to id={self.advocacy_campaigns[adv_camp['title']]}")
+                self.advocacy_campaigns[adv_camp["title"]] = adv_camp["identifiers"][0].split(':')[1]
```

Comment:
You could extract this session to another method to control better your context

Reply:
i need th current page, so i will still need this context and not achieve the goal of add another method